app/pages/lab/workflows.jsx

```diff
@@ -7,116 +7,161 @@ import WorkflowCreateForm from './workflow-create-form';
 import LoadingIndicator from '../../components/loading-indicator';
 import Paginator from '../../talk/lib/paginator';
 
-const WorkflowsPage = (props) => {
-  const renderWorkflowTable = ((workflow) => {
+class WorkflowsPage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleWorkflowSettingChange = this.handleWorkflowSettingChange.bind(this);
+    this.state = {
+      active: false,
+      error: null
+    };
+  }
+
+  handleWorkflowSettingChange(workflow, e) {
+    const defaultWorkflow = (this.props.configuration) ? this.props.configuration.default_workflow : null;
+    const checked = e.target.checked;
+    workflow.update({ active: checked }).save()
+      .catch(error => console.log(error))
+      .then(() => {
+        if (!workflow.active && workflow.id === defaultWorkflow) {
+          this.props.project.update({ 'configuration.default_workflow': null });
+          this.props.project.save();
+        }
+      })
+      .then(() => this.setState({ active: workflow.active }));
+  }
+
+  renderWorkflowTable(workflow) {
     const progressPercentage = workflow.completeness * 100;
     return (
       <tr key={workflow.id}>
         <td>
-          <Link key={workflow.id} to={props.labPath(`/workflows/${workflow.id}`)}  activeClassName="active">
+          <Link
+            key={workflow.id}
+            to={this.props.labPath(`/workflows/${workflow.id}`)}
+            activeClassName="active"
+          >
             {workflow.display_name}
-            {(props.project.configuration && workflow.id === props.project.configuration.default_workflow) && (
+            {(this.props.project.configuration && workflow.id === this.props.project.configuration.default_workflow) && (
               <span title="Default workflow">{' '}*{' '}</span>
             )}
           </Link>
         </td>
         <td>
           {`${progressPercentage.toFixed(0)} % Complete`}
         </td>
+        <td>
+          <label htmlFor="active">
+            <input
+              checked={workflow.active}
+              id="active"
+              onChange={this.handleWorkflowSettingChange.bind(this, workflow)}
+              type="checkbox"
+            />
+            Active
+          </label>
+        </td>
       </tr>
     );
-  });
+  }
 
-  const renderWorkflowList = ((workflow) => {
+  renderWorkflowList(workflow) {
```

Comment:
I'm not sure how the DragReorderable component would handle it, but also this could possibly be its own functional component.

Reply:
`DragReorderable` breaks when trying to pass a component to it, so I left `renderWorkflowList` alone given that it's not large.